### PR TITLE
Add Search Back Pressure PA metrics

### PR DESCRIPTION
**Pull request by ruai0511**
### Description
Added search back pressure related metrics in PA documentation.

### Issues Resolved
Update documentation

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by khushbr** [comment]
@Naarcha-AWS  Can we please prioritize this? This is for a `v2.11.0` feature.

**Comment by hdhalter** [comment]
@khushbr - Is there an associated development issue or PR related to this so I can get some context? Thanks.

**Comment by khushbr** [comment]
@ruai0511 Can you please answer @hdhalter 's question here? Thanks!

**Comment by ruai0511** [comment]
@hdhalter This is the PR related to search back pressure PA metrics: https://github.com/opensearch-project/performance-analyzer/pull/483

**Comment by hdhalter** [comment]
The table is not aligned in this file. Not your fault - it was already messed up. Let me see if I can fix it.

**Comment by hdhalter** [comment]
I fixed the table issues by reformatting the table. I pulled out the middle row, making the dimensions headings of the table. Let me know if this makes sense, is easier to read, and is technically correct. Thanks! https://github.com/opensearch-project/documentation-website/pull/5182.

File: _monitoring-your-cluster/pa/reference.md [hunk 1]
```diff
@@ -651,6 +651,140 @@ This list is extensive. We recommend using Ctrl/Cmd + F to find what you're look
...
+      </td>
+      <td rowspan="22">NodeID, searchbp_mode
+      </td>
+      <td>The number of tasks marked for cancellation, on search shard task level.
+      </td>
+    </tr>
+    <tr>
+      <td>searchbp_shard_stats_limitReachedCount
+      </td>
+      <td>The number of times when the number of tasks eligible for cancellation exceeded the set cancellation threshold, on search shard task level.
```
Comment (khushbr) [inline _monitoring-your-cluster/pa/reference.md:665]: The number of times when **cancellable task total(sum?)** exceeded the set cancellation threshold, on search shard task level.

File: _monitoring-your-cluster/pa/reference.md [hunk 2]
```diff
@@ -651,6 +651,140 @@ This list is extensive. We recommend using Ctrl/Cmd + F to find what you're look
...
+    <tr>
+      <td>searchbp_shard_stats_resource_heap_usage_currentMax
+      </td>
+      <td>The maximum heap usage for tasks currently running, on search shard task level.
+      </td>
+    </tr>
+    <tr>
+      <td>searchbp_shard_stats_resource_heap_usage_rollingAvg
+      </td>
+      <td> The rolling average heap usage for n most recent tasks, on search shard task level. The default value for n is 100.
```
Comment (khushbr) [inline _monitoring-your-cluster/pa/reference.md:683]: Can we highlight `n` here ? Either uppercase and/or special formatting.

File: _monitoring-your-cluster/pa/reference.md [hunk 3]
```diff
@@ -651,6 +651,140 @@ This list is extensive. We recommend using Ctrl/Cmd + F to find what you're look
...
+      </td>
+    </tr>
+    <tr>
+      <td>searchbp_shard_stats_resource_cpu_usage_cancellationCount
+      </td>
+      <td>he number of tasks marked for cancellation because of excessive CPU usage since the node last restarted, on search shard task level.
```
Comment (khushbr) [inline _monitoring-your-cluster/pa/reference.md:689]: Typo: Missing `T` for the beginning `The`

File: _monitoring-your-cluster/pa/reference.md [hunk 4]
```diff
@@ -651,6 +651,140 @@ This list is extensive. We recommend using Ctrl/Cmd + F to find what you're look
...
+      </td>
+    </tr>
+    <tr>
+      <td>searchbp_shard_stats_resource_cpu_usage_currentMax
+      </td>
+      <td>The maximum CPU time for all tasks currently running on the node, on search shard task level.
```
Comment (khushbr) [inline _monitoring-your-cluster/pa/reference.md:695]: `CPU time`, Is this `time` or `usage` ?

**Comment by khushbr** [comment]
@kaushalmahi12 Can you clarify if the cpu is measured in terms of elapsed time or utilization ? For reference `searchbp_shard_stats_resource_cpu_usage_currentMax` is described as `The maximum CPU time for all tasks currently running on the node, on search shard task level`, so is the more accurate description in terms of *CPU time* or *CPU utilization* ?

File: _monitoring-your-cluster/pa/reference.md (hunk 4 above)
Comment (ruai0511) [reply _monitoring-your-cluster/pa/reference.md:695]: I was referring to this [page](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/search-backpressure/) when writing. It has CPU time on it. 
Screenshot 2023-10-13 at 12 56 16 PM

Comment (kaushalmahi12) [reply _monitoring-your-cluster/pa/reference.md:695]: It is time in nanoseconds
```
public TaskResourceUsageTracker.Stats stats(List<? extends Task> activeTasks) {
        long currentMax = activeTasks.stream().mapToLong(t -> t.getTotalResourceStats().getCpuTimeInNanos()).max().orElse(0);
        long currentAvg = (long) activeTasks.stream().mapToLong(t -> t.getTotalResourceStats().getCpuTimeInNanos()).average().orElse(0);
        return new Stats(getCancellations(), currentMax, currentAvg);
    }
```

**Comment by hdhalter** [comment]
Closing in favor of https://github.com/opensearch-project/documentation-website/pull/5182. Note that 5182 was released without the new metrics which are being added in 2.12.

**Pull request closed without merging**